public/app/features/plugins/extensions/getPluginExtensions.ts

```diff
@@ -131,9 +164,14 @@ export const getPluginExtensions: GetExtensions = ({ context, extensionPointId,
   return { extensions };
 };
 
-function getLinkExtensionOverrides(pluginId: string, config: PluginExtensionLinkConfig, context?: object) {
+function getLinkExtensionOverrides(
+  pluginId: string,
+  config: PluginExtensionLinkConfig,
+  isAppOpened: () => boolean,
+  context?: object
+) {
   try {
-    const overrides = config.configure?.(context);
+    const overrides = config.configure?.(context, { isAppOpened });
```

Comment:
I am not sure about this one. Although not impossible, but it's also not simple to tell if an app (or any part of an app) is visible at a given time. The extension point that defines the `context` on the other hand could know about this and share it using the `context` - am I missing something?

Reply:
Yeah I have to think about also in context of what you said here:
> if an app plugin is registering itself (or any of its pages) under somewhere else in the nav tree, then the URL won't tell you if the page is served by an app plugin.

Like I assume there are 2 cases. One is we have to load the plugin so we need to know the plugin id and we can save it in the store to be used for this. Other is core functionality where I assume we can just add something to each route manually which would something like
```
useEffect( () => {
  markAppRender('dashbaords');
  return () => markUnload('dashboards');
}, [])
```

or something like that.

The context thing though is something that the app provides manually. Meaning that app developer would either have to pass it (and not forget about it), or we would still need some more automatic way to do this (similar to what I am proposing higher up) and make it automatically propagate into the extension context but at that point whether it's in the context or in additional helpers probably does not make that much difference.

Reply:
Same as https://github.com/grafana/grafana/pull/91648#discussion_r1713219200. 

Exactly, I was also thinking about something like this, where we keep an internal state somewhere of what app plugins are currently rendered (e.g. in the AppRootPage). But I think that doesn't necessarily need to be part of this PR, it's something that we can tackle later on.